### PR TITLE
emilua: init at 0.4.3

### PR DESCRIPTION
**Pull request by manipuladordedados**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/interpreters/emilua/default.nix [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    "-Dversion_suffix=-nix1"
+    "-Denable_http=true"
+    "-Denable_file_io=true"
+    "-Denable_io_uring=true"
+    "-Denable_linux_namespaces=true"
+    "-Denable_tests=true"
+    "-Denable_manpages=true"
+  ];
+
+  postUnpack = ''mv emilua-http emilua/
```
Comment (AndersonTorres) [inline pkgs/development/interpreters/emilua/default.nix:96]: Can you please indent that block of code?

Also this is ugly as hell. Read the code below and try to adapt it.

https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/window-managers/cardboard/default.nix

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/development/interpreters/emilua/default.nix (hunk 1 above)
Comment (manipuladordedados) [reply pkgs/development/interpreters/emilua/default.nix]: Done, thanks.

**Review by manipuladordedados — COMMENTED** [review]
Fixed

File: pkgs/development/interpreters/emilua/default.nix [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+, asciidoctor
+}:
+
+let
+  emilua-http-wrap = fetchFromGitHub {
+      owner = "BoostGSoC14";
+      repo = "boost.http";
+      rev = "93ae527c89ffc517862e1f5f54c8a257278f1195";
+      name = "emilua-http";
+      sha256 = "MN29YwkTi0TJ2V+vRI9nUIxvJKsG+j3nT3o0yQB3p0o=";
```
Comment (AndersonTorres) [inline pkgs/development/interpreters/emilua/default.nix:30]:
```suggestion
      hash = "sha256-MN29YwkTi0TJ2V+vRI9nUIxvJKsG+j3nT3o0yQB3p0o=";
```

File: pkgs/development/interpreters/emilua/default.nix [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+      fetchSubmodules = false;
```
Comment (AndersonTorres) [inline pkgs/development/interpreters/emilua/default.nix:31]:
```suggestion
```

This is the default already.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Comment by AndersonTorres** [comment]
Fixup the three latest commits into one.

**Pull request closed without merging**